lib/ansible/runner/response_async.rb

```diff
@@ -54,6 +54,25 @@ def dump
         }
       end
 
+      # Waits for the async process to complete or hit the given timeout
+      #
+      # @param timeout [Integer, ActiveSupport::Duration] Number of seconds to wait for the process to complete
+      # @return [Ansible::Runner::Response] Response object with all details about the Ansible run
+      def wait(timeout)
+        result = nil
+        # Poll once per second until complete
+        1.upto(timeout) do |t|
+          result = response
+          result ? break : sleep(1)
+        end
+        # If the process is still running, then stop it
+        if result.nil?
+          stop
```

Comment:
I could see wanting to wait X seconds for this to complete but _not_ want to stop it, could we pull this out and return `nil` if it timed out or something?  That way it is up to the caller if they want to kill the run on timeout or not.

Reply:
I think I want to leave this as is, because I the interface for wait is to return a response object (i.e. turn a response_async object into a response object.